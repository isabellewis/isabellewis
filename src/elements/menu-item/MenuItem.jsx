import React, { PureComponent } from 'react';
import styled from 'styled-components';

const MenuItemWrapper = styled.li`
	color: #001;
	cursor: pointer;
	display: block;
	font-family: 'Archivo Narrow', sans-serif;
	font-size: 15px;
	list-style-type: none;
	padding: 20px 10px;
	transition: color .5s;

	:hover {
		color: #78cedb;
	}

	@media only screen and (min-width: 576px) {
		display: inline-block;
		padding: 0 30px;
	}
`;


export class MenuItem extends PureComponent {
	constructor(props) {
		super(props);

		this.menuItemScroll = this.menuItemScroll.bind(this);
	}

	menuItemScroll () {
		const {
			menuItemLocation
		} = this.props;

		document.getElementById(menuItemLocation) &&
		document.getElementById(menuItemLocation).scrollIntoView({
			behavior: 'smooth'
		});
	};

	render () {
		const {
			menuItemText,
		} = this.props;
	
		return (
			<MenuItemWrapper onClick={this.menuItemScroll} onClick = {this.menuItemScroll}>
				<a>
					{ menuItemText }
				</a>
			</MenuItemWrapper>
		);
	}
}	
