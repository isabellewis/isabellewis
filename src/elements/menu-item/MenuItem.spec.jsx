import React from 'react';
import { shallow } from 'enzyme';
import { MenuItem } from './MenuItem';

describe('MenuItem', () => {
	let component;

	beforeEach(() => {
		component = shallow(<MenuItem menuItemText='MenuItem test text' menuItemLocation='aboutPanel' />);
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});