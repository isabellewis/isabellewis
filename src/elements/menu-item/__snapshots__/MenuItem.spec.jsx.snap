// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    menuItemLocation="aboutPanel"
    menuItemText="MenuItem test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "MenuItem test text",
      "className": "short-copy-text",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "MenuItem test text",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "MenuItem test text",
        "className": "short-copy-text",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "MenuItem test text",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
